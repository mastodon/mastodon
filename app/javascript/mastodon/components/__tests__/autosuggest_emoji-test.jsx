import renderer from 'react-test-renderer';

import AutosuggestEmoji from '../autosuggest_emoji';

describe('<AutosuggestEmoji />', () => {
  it('renders native emoji', () => {
    const emoji = {
      native: 'ðŸ’™',
      colons: ':foobar:',
    };
    const component = renderer.create(<AutosuggestEmoji emoji={emoji} />);
    const tree      = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders emoji with custom url', () => {
    const emoji = {
      custom: true,
      imageUrl: 'http://example.com/emoji.png',
      native: 'foobar',
      colons: ':foobar:',
    };
    const component = renderer.create(<AutosuggestEmoji emoji={emoji} />);
    const tree      = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
