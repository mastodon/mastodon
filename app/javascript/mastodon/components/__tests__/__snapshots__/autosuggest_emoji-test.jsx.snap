// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutosuggestEmoji /> renders emoji with custom url 1`] = `
<div
  className="autosuggest-emoji"
>
  <img
    alt="foobar"
    className="emojione"
    src="http://example.com/emoji.png"
  />
  <div
    className="autosuggest-emoji__name"
  >
    :foobar:
  </div>
</div>
`;

exports[`<AutosuggestEmoji /> renders native emoji 1`] = `
<div
  className="autosuggest-emoji"
>
  <img
    alt="ðŸ’™"
    className="emojione"
    src="/emoji/1f499.svg"
  />
  <div
    className="autosuggest-emoji__name"
  >
    :foobar:
  </div>
</div>
`;
